hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsHttpOperation.java

```diff
@@ -85,6 +88,7 @@ public class AbfsHttpOperation implements AbfsPerfLoggable {
   private long sendRequestTimeMs;
   private long recvResponseTimeMs;
   private boolean shouldMask = false;
+  private boolean expect100failureReceived = false;
```

Comment:
Though the significance of the bug is seen in 100-Continue flow, the state of the previously established connection is the reason to prevent getHeaderField* APIs. Would suggest renaming this field to connectionDisconnectedOnError and also add a code comment that this is a workaround for JDK bug with link.

Reply:
I believe, getOutputStream can throw only two major types of exception:
1. Expect-100 error
2. Other IOException
In case of other IOException, we should immediately throw back to AbfsRestOperation to do the retry. We can take responseCode only in case of Except-100 error, because there was a valid server call made. In other IOException, they can contain, CT, connection-reset etc.

Reply:
Have taken it. Refactored the variable name.